src/InitializeDatabaseScript.py

```diff
@@ -0,0 +1,201 @@
+    def connect(self):
+        try:
+            # Get the server params from Database.ini
+            params = config.config('postgres_server')
+            # Create a connection to the posgresql database server
+            self.db_server_conn = psycopg2.connect(**params)
+
+            self.db_cursor = self.db_server_conn.cursor()
+            # Print PostgreSQL Connection properties
+            print(self.db_server_conn.get_dsn_parameters(), "\n")
+
+            # Print PostgreSQL version
+            # possibly do not need class based cursor so this might need to be removed.
+            # TODO
+
+            self.db_cursor.execute("SELECT version();")
+            record = self.db_cursor.fetchone()
+            print("You are connected to - ", record, "\n")
+
+        except (Exception, psycopg2.Error) as e:
+            self.debugLogFile.writeToFile("Error while connecting to PostgreSQL " + str(e))
+            # raise Exception("Error while connecting to PostgreSQL " + str(e))
+
+            # print("Error while connecting to PostgreSQL", error)
+
+    # Disconnect is its own function so that the connection to the
+    # databases server can remain open while the database and table are being
+    # queried.
+    def disconnect(self):
+        if self.db_server_conn:
+            self.db_cursor.close()
+            self.db_server_conn.close()
+            print("PostgreSQL connection is closed")
+
+    def create_database(self):
+        if self.db_server_conn:
+            try:
+                self.database_params = config.config('postgres_db')
+                self.db_server_conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+                database_name = self.database_params.get('database')
+                sqlCreateDatabase = "create database " + database_name + ""
+                self.db_cursor.execute(sqlCreateDatabase)
+
+            except (Exception, psycopg2.Error) as e:
+                self.debugLogFile.writeToFile("Could not create Database " + str(e))
+
+    def create_table(self):
+        # need to connect to the database that I am actually creating the table in.
+        conn = None
+        cursor = None
+        try:
+            # connect to the database on the postgres server.
+            conn = psycopg2.connect(**self.database_params)
+            # conn = self.database_connect()
+            # once we are connected to the database that we created on the postgreSQL server
+            # we can create a table in that database.
+            cursor = conn.cursor()
+            # self.bug_table_name =
+
+            # basic Table
+            table = ("CREATE TABLE " + self.bug_table_name + """ (
+                        bug_title text NULL,
+                        bug_description text NULL,
+                        bug_traceback_info text NULL,
+                        bug_resolved BOOL NULL,
+                        PRIMARY KEY (bug_title)
```

Comment:
I think its safe to remove "bug" within all of these names and having bug in just the title of the table should be sufficient

Reply:
Done